src/presentation/test/integration/registration/address/choose-principal-place-of-business-address.test.ts

```diff
@@ -0,0 +1,130 @@
+import request from "supertest";
+import enTranslationText from "../../../../../../locales/en/translations.json";
+import cyTranslationText from "../../../../../../locales/cy/translations.json";
+import app from "../../app";
+import {
+  CHOOSE_PRINCIPAL_PLACE_OF_BUSINESS_ADDRESS_URL
+} from "presentation/controller/addressLookUp/url";
+import { GENERAL_PARTNERS_URL } from "../../../../controller/registration/url";
+import { getUrl, setLocalesEnabled, testTranslations } from "../../../utils";
+import { appDevDependencies } from "config/dev-dependencies";
+import * as config from "config";
+import AddressPageType from "presentation/controller/addressLookUp/PageType";
+
+describe("Choose Principal Place Of Business Address Page", () => {
+  const URL = getUrl(CHOOSE_PRINCIPAL_PLACE_OF_BUSINESS_ADDRESS_URL);
+  const REDIRECT_URL = getUrl(GENERAL_PARTNERS_URL);
+
+  beforeEach(() => {
+    setLocalesEnabled(false);
+    appDevDependencies.addressLookUpGateway.setError(false);
+    appDevDependencies.cacheRepository.feedCache({
+      [`${config.APPLICATION_CACHE_KEY_PREFIX_REGISTRATION}registered_office_address`]:
```

Comment:
Will update this, it is breaking tests atm

Reply:
Looked this is used by the controller for the address lookup it might need renaming.

Reply:
can you add an if statement into the address controller here https://github.com/companieshouse/limited-partnerships-web/blob/main/src/presentation/controller/addressLookUp/Controller.ts#L250, to say, if pagetype == your page, use a different cache key for the address?

Reply:
Added these checks into the controller.